### PR TITLE
Made docker file a whole lot smaller! (67% to be exact)

### PR DESCRIPTION
**Pull request by nivong**
## Short Description:
Rebased the docker to use alpine python build since it's a lot lighter and easier to download and handle on low connections speed

### CODE REVIEW

**Comment by nivong** [comment]
please wait for merge trying to optimize it even more

**Comment by nivong** [comment]
@TheSavior save to merge now :)
if someone else has an idea on how to make it even smaller let me know.

**Comment by helloj** [comment]
Please keep the following `ENTRYPOINT` code
```
ENTRYPOINT ["python", "pokecli.py"]
```
In order to execute docker image and pass parameters, such as `ptc, username, password`

**Comment by th3w4y** [comment]
@nivong isn't this a bit odd? why would one run setup.sh -i during docker build. why would i save my username and password in the docker image?

**Comment by th3w4y** [comment]
@mjmadsen: this creates some security concerns for me if the username and password are in the docker image.

**Comment by nivong** [comment]
Sorry it's not the file that is smaller it's the installation.  Now it's 800 something MB. This puts it down to 299 MB by using alpine Linux. But I'm new to docker so if there is a better way of streamlining the whole thing please do! 
I used setup.sh because installation instructions might be updated in the feature to not have to edit everything I used the setup.sh 
And yes you won't be promoted ;)  tested that out ofc

**Comment by nivong** [comment]
Good point about the entry point will change that asap when I'm back in the office (sorry about the stupid close fat fingers)

**Comment by th3w4y** [comment]
@nivong kudos for using alpine that part i totally agree and support 
my 2 cents were regarding the readability and transparency of what the Dockerfile does

**Comment by nivong** [comment]
Thanks will clean it up a bit. Could you perhaps contribute too? You can pr to the dev branch in my fork to update it here. Like I said I'm totally new at docker and not Linux. So if you have a better idea I'm all ears and eyes!

**Comment by nivong** [comment]
@th3w4y updated, is this more like it?

**Comment by th3w4y** [comment]
@nivong 

looks good
but check below version for the ADD directly from github
that way if user has any file modified they don't get pulled in by docker build.

```
FROM python:2.7.12-alpine

RUN apk add --update --no-cache alpine-sdk bash wget git

ADD https://github.com/PokemonGoF/PokemonGo-Bot/archive/dev.tar.gz .
RUN tar -zxvf dev.tar.gz && mv PokemonGo-Bot-dev /usr/src/app && rm dev.tar.gz

WORKDIR /usr/src/app
VOLUME ["/usr/src/app/configs", "/usr/src/app/web"]

ARG timezone=Etc/UTC
RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone

#setup the bot
RUN cd /tmp && wget http://pgoapi.com/pgoencrypt.tar.gz \
    && tar zxvf pgoencrypt.tar.gz \
    && cd pgoencrypt/src \
    && make \
    && cp libencrypt.so /usr/src/app/encrypt.so \
    && cd /tmp \
    && rm -rf /tmp/pgoencrypt*

ENV LD_LIBRARY_PATH /usr/src/app

RUN ln -s /usr/include/locale.h /usr/include/xlocale.h
RUN pip install --no-cache-dir -r requirements.txt

#remove unused stuff
RUN apk del alpine-sdk\
  && apk cache clean\
  && rm -rf /var/cache/apk/*

ENTRYPOINT ["python", "pokecli.py"]
```

**Comment by nivong** [comment]
no idea what it does but you have more knowledge than me ;) added it!

**Comment by th3w4y** [comment]
@nivong check https://github.com/nivong/PokemonGo-Bot/pull/4

**Comment by th3w4y** [comment]
@nivong thanks

File: README.md [hunk 1]
```diff
@@ -68,14 +68,7 @@ If you do not want any data to be gathered, you can turn off this feature by set
...
 
-## Credits
-- [tejado](https://github.com/tejado) many thanks for the API
-- [U6 Group](http://pgoapi.com) for the U6
-- [Mila432](https://github.com/Mila432/Pokemon_Go_API) for the login secrets
-- [elliottcarlson](https://github.com/elliottcarlson) for the Google Auth PR
-- [AeonLucid](https://github.com/AeonLucid/POGOProtos) for improved protos
-- [AHAAAAAAA](https://github.com/AHAAAAAAA/PokemonGo-Map) for parts of the s2sphere stuff
-- [breeze ro] for some of that map stuff
+## [Credits](https://github.com/PokemonGoF/PokemonGo-Bot/blob/master/CONTRIBUTORS.md)
```
Comment (th3w4y) [inline README.md]: @nivong why the Credits change?

Comment (nivong) [inline README.md]: aren't they in the credits? What is the difference in credits and contribution anyway?

File: Dockerfile [hunk 2]
```diff
@@ -1,14 +1,18 @@
-FROM python:2.7
+FROM python:2.7.12-alpine
+
+RUN apk add --update --no-cache alpine-sdk wget git
```
Comment (nivong) [inline Dockerfile]: @th3w4y do you think we still need git and wget? If not we should remove it to keep it clean.

**Comment by th3w4y** [comment]
@nivong: before you take out **wget** you would have to modify the Dockerfile like this:


```
-RUN cd /tmp && wget http://pgoapi.com/pgoencrypt.tar.gz \
-      && tar zxvf pgoencrypt.tar.gz \
+ADD http://pgoapi.com/pgoencrypt.tar.gz /tmp/
+RUN cd /tmp && tar zxvf pgoencrypt.tar.gz \

```

as for git i don't  know if is used

File: Dockerfile (hunk 2 above)
Comment (th3w4y) [inline Dockerfile]: @nivong: before you take out wget you would have to modify the Dockerfile like this:

-RUN cd /tmp && wget http://pgoapi.com/pgoencrypt.tar.gz \
-      && tar zxvf pgoencrypt.tar.gz \
+ADD http://pgoapi.com/pgoencrypt.tar.gz /tmp/
+RUN cd /tmp && tar zxvf pgoencrypt.tar.gz \

as for git i don't know if is used

**Comment by nivong** [comment]
can you do a PR to my fork :) ? but swampt today

**Comment by th3w4y** [comment]
@nivong https://github.com/nivong/PokemonGo-Bot/pull/6

**Comment by th3w4y** [comment]
@nivong please pull in also https://github.com/nivong/PokemonGo-Bot/pull/7 otherwise the suggestion i have with pulling the archive from github.com and the fact that i hardcoded dev will not be compatible with the master branch

**Comment by nivong** [comment]
check!

**Comment by nivong** [comment]
@th3w4y this comes up on testing the docker build:
```
Step 21 : RUN apk del alpine-sdk   && rm -rf /var/cache/apk/*
 ---> Running in 2ba527e2474e
WARNING: Ignoring APKINDEX.167438ca.tar.gz: No such file or directory
WARNING: Ignoring APKINDEX.a2e6dac0.tar.gz: No such file or directory
Removing intermediate container 2ba527e2474e
The command '/bin/sh -c apk del alpine-sdk   && rm -rf /var/cache/apk/*' returned a non-zero code: 1
```
Why is this a problem out of the blue? Worked before....

**Comment by th3w4y** [comment]
@nivong checking now...
`docker build https://raw.githubusercontent.com/nivong/PokemonGo-Bot/691037348b7d40649f8c507e73a12334854b5526/Dockerfile`
... still building

**Comment by th3w4y** [comment]
@nivong 
I found the cullprit that step si ruined twice....

```

#remove unused stuff
RUN apk del alpine-sdk\
  && rm -rf /var/cache/apk/*

ENTRYPOINT ["python", "pokecli.py"]

#remove unused stuff
RUN apk del alpine-sdk \
  && rm -rf /var/cache/apk/*

ENTRYPOINT ["python", "pokecli.py"]
```

**Comment by nivong** [comment]
lol how did that happen, changed it thanks for checking, if you have more please do a PR i am out of the office again

**Comment by th3w4y** [comment]
@nivong sorry was my bad in commit 215832021111bd08038b337487b09d68e73787ce   i copy paste from my file over the text in the github.com editor and probably some lines remain unselected when pasting over.

**Comment by th3w4y** [comment]
@nivong now all looks good  👍 

```
14:06 $ docker build https://raw.githubusercontent.com/nivong/PokemonGo-Bot/8832b16f610bbc7ea7cbce10b8c682450a6d05da/Dockerfile
Downloading build context from remote url: https://raw.githubusercontent.com/nivong/PokemonGo-Bot/8832b16f610bbc7ea7cbce10b8c682450a6d05da/Dockerfile  1.56 kB
Sending build context to Docker daemon 3.584 kB
Step 1 : FROM python:2.7.12-alpine
 ---> 67fa4e31547b
Step 2 : RUN apk add --update --no-cache alpine-idk git
 ---> Using cache
 ---> a4ae9dd908be
Step 3 : ARG BUILD_BRANCH
 ---> Using cache
 ---> 321d0cbbdaf6
Step 4 : ENV BUILD_BRANCH ${BUILD_BRANCH:-master}
 ---> Using cache
 ---> 2dcad1163c8d
Step 5 : ARG BUILD_REPO
 ---> Using cache
 ---> b4eaf959943c
Step 6 : ENV BUILD_REPO ${BUILD_REPO:-PokemonGoF/PokemonGo-Bot}
 ---> Using cache
 ---> 12cc3e912e5d
Step 7 : LABEL build_repo $BUILD_REPO build_branch $BUILD_BRANCH
 ---> Using cache
 ---> 06c9d4154b8c
Step 8 : ADD https://github.com/$BUILD_REPO/archive/$BUILD_BRANCH.tar.gz .
Downloading [==================================================>] 3.521 MB/3.521 MB
 ---> Using cache
 ---> ec957c9739f9
Step 9 : RUN tar -zxvf $BUILD_BRANCH.tar.gz && mv PokemonGo-Bot-* /usr/src/app && rm $BUILD_BRANCH.tar.gz
 ---> Using cache
 ---> 047661f5c49e
Step 10 : WORKDIR /usr/src/app
 ---> Using cache
 ---> c39fa7fe8bbc
Step 11 : VOLUME /usr/src/app/configs /usr/src/app/web
 ---> Using cache
 ---> 47815483d5f0
Step 12 : ARG timezone=Etc/UTC
 ---> Using cache
 ---> 9ca574da35f9
Step 13 : RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
 ---> Using cache
 ---> 2bc5d76191de
Step 14 : ADD http://pgoapi.com/pgoencrypt.tar.gz /tmp/
Downloading [==================================================>] 243.6 kB/243.6 kB
 ---> Using cache
 ---> dc84fb88af8b
Step 15 : RUN cd /tmp && tar zxvf pgoencrypt.tar.gz     && cd pgoencrypt/src     && make     && cp libencrypt.so /usr/src/app/encrypt.so     && cd /tmp     && rm -rf /tmp/pgoencrypt*
 ---> Using cache
 ---> 861c0b5a4784
Step 16 : ENV LD_LIBRARY_PATH /usr/src/app
 ---> Using cache
 ---> eb78d51f0398
Step 17 : RUN ln -s /usr/include/locale.h /usr/include/xlocale.h
 ---> Using cache
 ---> fa24a7864b5c
Step 18 : RUN pip install --no-cache-dir -r requirements.txt
 ---> Using cache
 ---> 0e858ca5141c
Step 19 : RUN apk del alpine-sdk  && rm -rf /var/cache/apk/*
 ---> Using cache
 ---> 7f5393c52c4e
Step 20 : ENTRYPOINT python pokecli.py
 ---> Using cache
 ---> d483306518cb
Successfully built d483306518cb
(anaconda3) (root) ✔ ~ 
14:06 $ echo $?
0
(anaconda3) (root) ✔ ~ 
14:06 $ 
```

**Comment by nivong** [comment]
Thanks for the merge @solderzzc and thanks @th3w4y for the help :)

**Pull request merged**